lib/publiccloud/provider.pm

```diff
@@ -708,19 +708,31 @@ sub terraform_param_tags
 =head2 get_terraform_output
 
 Query the terraform data structure in json format.
+Automatic search and set valid terraform directory on helper image.
 Input: <jq-query-format> string; <empty> = no query then full output of data structure.
 E.g: to get the VM instance name from json data structure, the call is: 
     get_terraform_output(".vm_name.value[0]");
 To get the complete output structure, the call is:
-    get_terraform_output();
+    get_terraform_output(".");
 
 =cut
 
 sub get_terraform_output {
     my ($self, $jq_query) = @_;
-    my $res = script_output("terraform output -no-color -json | jq -r '$jq_query' 2>/dev/null", proceed_on_failure => 1);
-    # jq 'null' shall return empty
-    return $res unless ($res =~ /^null$/);
+    # cuurent dir saved
+    my $dir = script_output("pwd");
+    # terraform state file search first matching
+    my $cd = script_output("find ~ -name \*.tfstate -exec dirname {} + -quit 2>/dev/null");
+    # state file missing: no data found
+    return "" if ($cd eq "");
+    # enter terraform dir. found
+    script_run("cd $cd") if ($cd ne $dir);
+    # get data
+    my $res = script_output("terraform output -no-color -json | jq -Mr '$jq_query'", proceed_on_failure => 1);
+    # restore saved dir
+    script_run("cd $dir") if ($cd ne $dir);
```

Comment:
```suggestion
    script_run("cd -");
```

Reply:
Sorry, about your suggestion:
- we move to _new_ dir. only when is `different` from the _current_ one; 
- but if _new_ destination is already the `same` as _current_ dir. **no cd done**, but in this case `cd -` would move to the `$OLDPWD` present `before entering` the routine, that is _not_ what we want.

Therefore I setcthat check.

Reply:
I could remove both those `if...` and always run `cd new` then `cd -` (because eventually 1st cd was the same , the next will have `$OLDPWD == $PWD` and work).
But in usch case we run 2 more unuseful commands.

Reply:
Why not use terraform's `--chdir` option?

Reply:
Pls see my replies in Slack thread, h16.22 and h17.09:
https://suse.slack.com/archives/C02CGKBCGT1/p1719241755418859?thread_ts=1719227385.937299&cid=C02CGKBCGT1

Reply:
It'd became a test of terraform application, out of scope of this PR.
Moreover there is no clear explaination what would be benefit and improvement in using it.

Reply:
> It'd became a test of terraform application.

We'd be using the terraform tool to achieve our goals, not testing it.  

> Moreover there is no clear explaination what benefit and improvement would be gained in using it.

There's a clear explanation.  A simple `terraform -chdir` would be **orders of magnitude** more cleaner.

Reply:
> We'd be using the terraform tool to achieve our goals, not testing it.

In fact, **again** as already exlpained, adding `-cdir` here is off-topic and even would force to do also more unuseful and out-scope tests here on `that` command, and therefore this `-chdir` proposal a.t.m. is not unseful here: _out of scope_.
Again, my goal in this PR is `AUTOMATIC FIND and validate` the **correct place where terraform data are located**, and NOT how do we set that place. Eventually, the discussiom on `-chdir` addition can be postponed to another time, after this work consolidated.

Reply:
> > We'd be using the terraform tool to achieve our goals, not testing it.
> 
> In fact, **again** as already exlpained, adding `-cdir` here is off-topic and even would force to do also more unuseful and out-scope tests here on `that` command, and therefore this `-chdir` proposal a.t.m. is not unseful here: _out of scope_. Again, my goal in this PR is `AUTOMATIC FIND and validate` the **correct place where terraform data are located**, and NOT how do we set that place. Eventually, the discussiom on `-chdir` addition can be postponed to another time/PR, after this work consolidated.

We're not testing `-chdir`.